src/main/java/org/dsa/iot/alarm/AlarmStreamer.java

```
@@ -0,0 +1,171 @@
+    private ActionResult request;
+    private Table table;
+    private LinkedList<AlarmRecord> updates = new LinkedList<>();
+
+    ///////////////////////////////////////////////////////////////////////////
+    // Constructors
+    ///////////////////////////////////////////////////////////////////////////
+
+    /**
+     *  Will set this as the close handler on the given request and will add and
+     *  remove itself from the given container.
+     *
+     *  @param container Where to add and remove this instnace.
+     *  @param initialSet Optional.
+     */
+    public AlarmStreamer(
+            Collection container,
+            ActionResult request,
+            AlarmCursor initialSet) {
+        request.setCloseHandler(this);
+        this.container = container;
+        synchronized (container) {
+            container.add(this);
+        }
+        this.request = request;
+        this.initialSet = initialSet;
+        request.setStreamState(StreamState.INITIALIZED);
+        this.table = request.getTable();
+        table.setMode(Table.Mode.STREAM);
+        table.sendReady();
+    }
+
+    ///////////////////////////////////////////////////////////////////////////
+    // Methods
+    ///////////////////////////////////////////////////////////////////////////
+
+    /**
+     * Force close the stream.
+     */
+    public void close() {
+        closed = true;
+    }
+
+    /**
+     * Does not return until there is a record, or the stream is closed.
+     *
+     * @return Possibly null if the stream is closed.
+     */
+    public AlarmRecord getNextUpdate() {
+        synchronized (updates) {
+            while (isValid() && (updates.size() == 0)) {
+                try {
+                    updates.wait(5000);
+                } catch (Exception ignore) {}
+                if (updates.size() > 0) {
+                    return updates.removeFirst();
+                }
+            }
+        }
+        return null;
```

should throw instead of returning null I think